t/t2800-jobs-cmd.t

```diff
@@ -82,7 +91,12 @@ test_expect_success 'submit jobs for job list testing' '
 	#  Run a job that we will end with a signal, copy its JOBID to both inactive and
 	#   failed and terminated lists.
 	#
-	jobid=`flux submit --wait-event=start sleep inf` &&
+	# N.B. sleepinf.sh and wait-event on job data to workaround
+	# rare job startup race.  See #5210
+	#
+	jobid=`flux submit ./sleepinf.sh` &&
```

Comment:
You need to keep the `--wait-event=start` option because the `guest.exec.eventlog` won't appear until the job is started.

Reply:
i think the job info module takes care of this.  We have a test for it in `t2231-job-info-eventlog-watch.t`

```
# In order to test watching a guest event log that does not yet exist,
# we will start a job that will take up all resources.  Then start
# another job, which we will watch and know it hasn't started running
# yet. Then we cancel the initial job to get the new one running.

test_expect_success 'job-info: generate jobspec to consume all resources' '
        flux run --dry-run -n4 -c2 sleep 300 > sleeplong-all-rsrc.json
'

test_expect_success NO_CHAIN_LINT 'flux job wait-event -p guest.exec.eventlog works (wait job)' '
```

Reply:
Oh, nice. I didn't realize that. Does it support this for the `guest.output` eventlog as well?

Reply:
I guess not:
```
$ flux job wait-event -v -p guest.output $(flux submit hostname) shell.init 
flux-job: eventlog path guest.output not found
```
Does that seem kind of confusing that job-info supports waiting for one standard eventlog but not the other?

Supporting this would really simplify support for watching the output eventlog.

Reply:
Yeah, I created an issue for it this morning:  https://github.com/flux-framework/flux-core/issues/5301

I believe the issue is b/c `guest.output` is created by the shell not the job-manager, so there's that window of time that the guest namespace exists but that specific eventlog doesn't.